### PR TITLE
Move metadatas used for proxy into common place

### PR DESCRIPTION
**Pull request by mathspanda**
Create a metadata library for istio proxy.

Fix issue #9710 

Signed-off-by: mathspanda <mathspanda826@gmail.com>

### CODE REVIEW

File: pkg/metadata/node/node.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	// By default its set to "" so that the proxy sees endpoints from the default unnamed network.
+	RequestedNetworkView = "REQUESTED_NETWORK_VIEW"
+
+	// UserSDS metadata decides whether to enable ingress gateway agent.
+	UserSDS = "USER_SDS"
+
+	// InstanceIPs metadata holds the IP addresses of the proxy used to identify it and its
+	// co-located service instances. In some cases, the host where the proxy and service instances
+	// reside may have more than one IP address. Example: "10.3.3.3,10.4.4.4".
+	InstanceIPs = "ISTIO_META_INSTANCE_IPS"
```
Comment (hzxuzhonghu) [inline pkg/metadata/node/node.go]: Why only this with `ISTIO_META_` prefix

**Comment by hzxuzhonghu** [comment]
/lgtm

@mathspanda  Could you also replace all metadata in test files?

**Comment by mathspanda** [comment]
> /lgtm
> 
> @mathspanda Could you also replace all metadata in test files?

Yeap. Have replaced all.

File: pkg/metadata/pilot/pilot.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	ProxyVersion = "ISTIO_PROXY_VERSION"
+
+	// ProxySHA metadata indicates the exact proxy sha, for debugging or version-specific configs.
+	ProxySHA = "ISTIO_PROXY_SHA"
+
+	// IstioVersion metadata indicates the exact build, for debugging.
+	IstioVersion = "ISTIO_VERSION"
+
+	// PodName metadata indicates the name of pod the proxy sidecars with.
+	PodName = "POD_NAME"
```
Comment (hzxuzhonghu) [inline pkg/metadata/pilot/pilot.go]: remove this, maybe this is legacy. I am sure it is useless now for pilot.

**Comment by hzxuzhonghu** [comment]
Thanks 

/lgtm

**Pull request closed without merging**